scripts/leodict.pl

```diff
@@ -53,41 +89,62 @@ ()
     print CLIENTCRAP draw_box($IRSSI{name}, $text, "help", 1);
 }
 
-
-
-sub get_page ($) {
-    my ($word) = @_;
-    return get('http://dict.leo.org/?search='.$word.'&relink=off');
-}
-
-sub get_words ($) {
-    my ($word) = @_;
-    my @translations;
-    my $data = get_page($word);
-    foreach (split(/\n/, $data)) {
-	if (/(\d+) search results/) {
-	    my $results = $1;
-	    foreach (split(/\<\/TR\>/)) {
-		my @trans;
-		foreach (split(/\<\/TD\>/)) {
-		    $_ =~ s/\<.*?\>//g;
-		    $_ =~ s/^ *//g;
-		    $_ =~ s/ *$//g;
-		    $_ =~ s/&nbsp;//g;
-		    $_ =~ s/^\t*//g;
-		    # Thanks to senneth
-		    $_ =~ s/Direct Matches//g;
-		    next if (/\d+ search results/);
-		    #print $_."\n" if (/\w/);
-		    push @trans, $_ if (/\w/);
-		}
-		if (scalar(@trans) == 2) {
-		    push @translations, \@trans;
+sub parser {
+    my @tbl;
+    my $headname;
+    my %ignore=(
+	'#Suchwort' => 1,
+	'Beispiele' => 1,
+	'Orthographisch ähnliche Wörter' => 1,
+	'Aus dem Umfeld der Suche' => 1,
+	'Forumsdiskussionen, die den Suchbegriff enthalten' =>1,
+	#'Substantive'
+	#'Verben'
+	#'Adjektive / Adverbien'
+	#'Phrasen'
+	'nope' => 1,
+    );
+    %fresult=();
+    while ( $ftext =~ m#(<table.*?</table>)#p) {
+	push @tbl,$1;
+	$ftext=$';
+    }
+    foreach my $t (@tbl ) {
+	# head
+	$headname='nope';
+	while ( $t =~ m#(<thead.*?</thead>)#p) {
+	    $t=$';
+	    $_=$1;
+	    $_=~s/<.*?>//g;
+	    $headname=$_;
+	}
+	next if (exists $ignore{$headname} );
+	# rows
+	my @rows=();
+	while ( $t =~ m#(<tr.*?</tr>)#p) {
+	    $t=$';
+	    my $r=$1;
+	    my @col=();
+	    my $c=0;
+	    while ( $r =~ m#(<td.*?</td>)#p ) {
+		$r=$';
+		$_=$1;
+		$_=~s/<.*?>//g;
+		if (length($_) >2) {
+		    $col[$c]=$_;
+		    $c++;
```

Comment:
Don't parse html with regex, there are html parsers for that
perlbot ┃ vague: Don't parse or modify html with regular expressions! See Mojo::DOM or HTML::TreeBuilder or suchlike. For a Feb-2014 coverage of main techniques see: http://is.gd/4FwOKu . If your response includes "that's overkill. i only want to..." you are wrong.           http://enwp.org/Chomsky_hierarchy and http://is.gd/Yk3uww for why not to use regex on HTML. | https://pbs.twimg.com/media/CgjfnfeW0AA0G5h.jpg